spec/datetime.html

```diff
@@ -1,352 +1,824 @@
+        1. Let _duration_ be ? ToTimeDurationLike(_duration_).
+        1. If _disambiguation_ is not *undefined*, then
+          1. Set _disambiguation_ to ? ToString(_disambiguation_).
+        1. Else
+          1. Set _disambiguation_ to `"constrain"`.
+        1. <mark>TODO</mark>
+      </emu-alg>
+    </emu-clause>
 
-  <emu-clause id=sec-temporal-civildatetime-withzone-alg>
-    <h1>CivilDateTime.withZone (timeZone[, options])</h1>
-    <p>The abstract operation plus takes as an argument a Time Zone String and an optional Object. It takes the current
-      CivilDateTime object, converts its values to Unix epoch time. It creates a new ZonedDateTime object.</p>
-    <p>When the withZone function is called, the following steps are taken:</p>
-    <emu-alg>
-      1. Let numberOfArgs be the number of arguments passed to this function call.
-      1. Assert: numberOfArgs > 1.
-      1. Assert: Type(timeZone) is String.
-      1. If options is supplied, assert: Type(options) Object.
-      1. If NewTarget is not undefined, then -- do something to convert. Get more input.
-    </emu-alg>
+    <emu-clause>
+      <h1>Temporal.DateTime.prototype.difference ( _otherDateTime_ )</h1>
+      <p>
+        The `difference` method takes one argument _otherDateTime_.
+        The following steps are taken:
+      </p>
+      <emu-alg>
+        1. Let _dateTime_ be the *this* value.
+        1. If Type(_dateTime_) is not Object or _dateTime_ does not have an [[InitializedTemporalDateTime]] internal slot, then
+          1. Throw a *TypeError* exception.
+        1. Let _otherDateTime_ be ? ToDateTime(_otherDateTime_).
+        1. If ! DateTimeCompare(_dateTime_, _otherDateTime_) &lt; 0, then
+          1. Let _greater_ be _otherDateTime_.
+          1. Let _smaller_ be _dateTime_.
+        1. Else,
+          1. Let _greater_ be _dateTime_.
+          1. Let _smaller_ be _otherDateTime_.
+        1. <mark>TODO</mark>
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause>
+      <h1>Temporal.DateTime.prototype.toString ( )</h1>
+      <p>
+        The following steps are taken:
+      </p>
+      <emu-alg>
+        1. Let _dateTime_ be the *this* value.
+        1. If Type(_dateTime_) is not Object or _dateTime_ does not have an [[InitializedTemporalDateTime]] internal slot, then
+          1. Throw a *TypeError* exception.
+        1. Return ! DateTimeToString(_dateTime_).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause>
+      <h1>Temporal.DateTime.prototype.toLocaleString ( [ _locales_ [ , _options_ ] ] )</h1>
+      <p>
+        The following steps are taken:
+      </p>
+      <emu-alg>
+        1. Let _dateTime_ be the *this* value.
+        1. If Type(_dateTime_) is not Object or _dateTime_ does not have an [[InitializedTemporalDateTime]] internal slot, then
+          1. Throw a *TypeError* exception.
+        1. If the implementation does not include the ECMA-402 Internationalization API, then
```

Comment:
I don’t think this is a realistic mechanism; instead, the fallback logic should be defined as an abstract operation, and 402 should override/replace that abstract operation.

Reply:
I'm not sure what you mean by "realistic" in this context. (I also notice that ECMA262 doesn't use an abstract operation.)

Reply:
Is there an existing place in 262 where 402 is mentioned in this way?

Reply:
No, neither have precedent. Let's keep this wording for now; we can do the more complicated thing when merging into the main specs.

Reply:
The more complicated thing will be required to pass editor review, required for stage 3, but that certainly doesn’t have to be in this PR.

Reply:
The current pattern is to have a dummy definition in 262 which is overridden in 402. I think we can work this out when we are editing the PRs to merge into the upstream specs.